### PR TITLE
Remove our perf job infrastructure

### PR DESCRIPTION
**Pull request by jaredpar**
This is not used or maintained and has not been for almost a year now.
Removing from our infrastructure. The new idea is to use the perf
infrastructure added to CLI.

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -446,53 +442,11 @@ function Test-Special() {
...
-        Test-PerfRun
-    }
     else {
         throw "Not a special test"
     }
 }
 
-function Test-PerfCorrectness() {
-    Run-MSBuild "Roslyn.sln" "/p:DeployExtension=$deployExtensions" -logFileName "RoslynPerfCorrectness"
-    Exec-Block { & ".\Binaries\$buildConfiguration\Exes\Perf.Runner\Roslyn.Test.Performance.Runner.exe" --ci-test } | Out-Host
```
Comment (jasonmalinowski) [inline build/scripts/build.ps1]: Are we deleting this too, or is it already deleted?

File: perf.groovy [hunk 2]
```diff
@@ -1,49 +0,0 @@
-// Groovy Script: http://www.groovy-lang.org/syntax.html
```
Comment (jasonmalinowski) [inline perf.groovy]: Note this file is explicitly called out in the netci configuration. Not sure if we have to delete it first?

File: build/scripts/build.ps1 (hunk 1 above)
Comment (jaredpar) [reply build/scripts/build.ps1]: Updated to delete

File: perf.groovy (hunk 2 above)
Comment (jaredpar) [reply perf.groovy]: https://github.com/dotnet/dotnet-ci/pull/994

**Pull request merged**